crates/vault-registry/src/types.rs

```diff
@@ -116,14 +116,21 @@ impl Default for VaultStatus {
     }
 }
 
+#[derive(Encode, Decode, Default, Clone, PartialEq)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct ReplaceRequest<PolkaBTC, DOT> {
```

Comment:
I'm not sure about moving this to the vault registry, it seems "dirty". Is there anyway we can still track `to_be_replaced_tokens` here and move `reserved_griefing_collateral` back into the `replace` pallet?

Reply:
The problem is that the redeem crate also needs access to this, since a `redeem` also decreases the to-be-replaced tokens (and it needs to release the collateral when it does so)

Reply:
My suggestion would be to remove r`eserved_griefing_collateral` actually, and only lock on accept

Reply:
As discussed offline, let's flatten this struct into the `Vault` type